test/e2e/storage/testsuites/provisioning.go

```diff
@@ -305,35 +305,78 @@ func (p *provisioningTestSuite) DefineTests(driver storageframework.TestDriver,
 					}
 					e2evolume.TestVolumeClientSlow(f, myTestConfig, nil, "", tests)
 				}
-				myTestCase.TestDynamicProvisioning()
+				t.TestDynamicProvisioning()
 			}(i)
 		}
 		wg.Wait()
 	})
 }
 
+// ProvisionStorageClass provisions a StorageClass from a spec, if the StorageClass already exists
+// then it returns it as a `computed` StorageClass, if it doesn't exist then it's created first
+// and then returned, if the spec is nil then we return the `default` StorageClass
+func ProvisionStorageClass(
+	client clientset.Interface,
+	class *storagev1.StorageClass,
+) (*storagev1.StorageClass, func()) {
+	gomega.Expect(client).NotTo(gomega.BeNil(), "ProvisionStorageClass.client is required")
+
+	var err error
+	var computedStorageClass *storagev1.StorageClass
+	var clearComputedStorageClass = func() {}
+	if class != nil {
+
+		computedStorageClass, err = client.StorageV1().StorageClasses().Get(context.TODO(), class.Name, metav1.GetOptions{})
+		if err == nil {
+			// skip storageclass creation if it already exists
+			ginkgo.By("Storage class " + computedStorageClass.Name + " is already created, skipping creation.")
+		} else {
+			ginkgo.By("Creating a StorageClass " + class.Name)
+			_, err = client.StorageV1().StorageClasses().Create(context.TODO(), class, metav1.CreateOptions{})
```

Comment:
Want to verify: Is there some case that different test share the same storageClass name, but the spec of the storageclass might be different. Need to verify this behavior.

Reply:
I did a search and all of the test cases inside testsuites create a StorageClass here https://github.com/kubernetes/kubernetes/blob/89d1e37ff116fbd5f41e7afee06af601ae611e66/test/e2e/storage/testsuites/provisioning.go#L144, this method is implemented by multiple drivers and all of them call https://github.com/kubernetes/kubernetes/blob/66fa2c49e28b9dcf478a87636971bc7d241b2849/test/e2e/storage/framework/driver_operations.go#L65 which creates a unique StorageClass name

TestDynamicProvisioning was always creating the StorageClass this way, what would we verify in this case?

Reply:
I thought we were saying the storageClass name is the same to different tests that is why storageClass might already exist when test tries to create it. If each storage class is unique for each test, it normally should not exist already?

Reply:
that's correct, if the storageclass already exists then the test could fail, I can change this to throw an exception instead

Reply:
One of the cases where the storage class already exists when it enters this method is in the v1beta1 case, the storage class is created outside this method and when this method is called then it's fetched from the cluster but as a v1 object, do you think that in the beta case I shouldn't use this method?